dog.md

```diff
@@ -0,0 +1,204 @@
+```
+
+File-able Fields
+----------------
+
+These parameters are controlled by governance:
+
+-   `vow`:  Debt Engine
+-   `hole`: Max DAI needed to cover debt+fees of active auctions
+-   `chop`: Liquidation Penalty
+-   `clip`: Liquidator
+
+```k
+    syntax DogAuthStep ::= "file" DogFile
+ // -------------------------------------
+
+    syntax DogFile ::= "vow" Address
+                     | "Hole" Rad
+                     | "hole" String Rad
+                     | "chop" String Wad
+                     | "clip" String Address
+
+ // -------------------------------------
+    rule <k> Dog . file vow VOW => . ... </k>
+         <dog-vow> _ => VOW </dog-vow>
+
+    rule <k> Dog . file Hole HOLE => . ... </k>
+         <dog-hole> _ => HOLE </dog-hole>
+
+    rule <k> Dog . file hole ILK HOLE => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... hole: (_ => HOLE) ) ... </dog-ilks>
+
+    rule <k> Dog . file chop ILK CHOP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... chop: (_ => CHOP) ) ... </dog-ilks>
+
+    rule <k> Dog . file clip ILK CLIP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: (_ => CLIP) ) ... </dog-ilks>
+```
+
+Dog Semantics
+-------------
+
+```k
+   syntax DogStep ::= "bark" String Address Address
+ // -------------------------
+   rule <k> Dog . bark ILK URN KPR
+   => #fun(ROOM
+   => #fun(DART_INITIAL
+   => #fun(DART_FINAL
+   => #fun(DINK
+   => #fun(DUE
+   => #fun(TAB
+   => call DOG_VAT . grab ILK URN CLIP DOG_VOW (wad(0) -Wad DINK) (wad(0) -Wad DART_FINAL)
+   ~> call DOG_VOW . fess DUE
+   ~> call CLIP . kick TAB DINK URN KPR
+   ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0   
+   )( DUE *Wad CHOP )
+   )( DART_FINAL *Wad Ray2Wad(RATE) )
+   )( (INK *Wad DART_FINAL) /Wad ART )
+   )( #if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Wad Ray2Wad(RATE)) <Wad Rad2Wad(DUST) ) #then ART #else DART_INITIAL #fi)
+   )( minWad( ART, ( ( Rad2Wad(ROOM) /Wad Ray2Wad(RATE)) /Wad CHOP) ) )
+   )( minRad( ( DOG_HOLE -Rad DOG_DIRT), (HOLE -Rad  DIRT) ) )
+   ... </k>
+   <dog-vow> DOG_VOW </dog-vow>
+   <dog-live> true </dog-live>
+   <dog-vat> DOG_VAT </dog-vat>
+   <vat-urns> ... { ILK, URN } |-> Urn( INK, ART) ... </vat-urns>
+   <dog-ilks> ... ILK |-> Ilk( ... clip: CLIP:ClipContract, chop: CHOP, hole: HOLE, dirt: DIRT ) ...</dog-ilks>
+   <vat-ilks> ... ILK |-> Ilk( ... rate: RATE, spot: SPOT, dust: DUST ) ... </vat-ilks>
+   <dog-dirt> DOG_DIRT </dog-dirt>
+   <dog-hole> DOG_HOLE </dog-hole>
+   requires ( ( SPOT >Ray ray(0) ) andBool ( (Wad2Ray(INK) *Ray SPOT ) <Ray (Wad2Ray(ART) *Ray RATE) ) )
+   andBool ( (DOG_HOLE >Rad DOG_DIRT ) andBool ( HOLE >Rad DIRT ) )
+
+
+   rule <k> ID:Int ~> emitBark ILK URN DINK DART DUE CLIP _ => emitBark ILK URN DINK DART DUE CLIP ID ... </k>
+   <vat-urns> ... { ILK, URN } |-> Urn( _, ART) ... </vat-urns>
+   <vat-ilks> ... ILK |-> Ilk( ... rate: RATE, spot: _, dust: DUST) ... </vat-ilks>
```

Comment:
Looks like some updates are missing here--`DOG_DIRT` needs to increase by `TAB`, as does the ilk-specific `DIRT`. Or maybe those belong in the rule above, but either way I don't them happening the way I see updates happening e.g. for `take`.